### DISCUSSION TITLE
'detached' window won't 'attach'

### DISCUSSION THREAD

**Discussion by g7kse** [foldynl/QLog#266] [General]
Probably something really daft but I managed to detach the dx cluster window and went to re attach it and can't seem to do it. Now its floating as a second window. 

I know its something really simple but I've had it floating for days now and can't seem to get it back...please someone put me out of my shame :-)

![image](https://github.com/foldynl/QLog/assets/13124178/f5c2ac90-e00b-43b3-b7fe-7389e297bc69)

**Comment by foldynl** [comment]
The attached animation may help.

![attach](https://github.com/foldynl/QLog/assets/942769/9d3a58c4-3ce9-4f63-9e23-0c03986f928e)

Otherwise, the steps to reattach a window should be this.

1) press the left mouse button above the upper edge of the window and hold the button
2) drag the window to the right side of Qlog. I see you have a CW Console there
3) Release the left button. The window should be attached to the highlighted part of the main window.

Hopefully it will help you.

**Comment by g7kse** [comment]
unfortunately not. Same behaviour with any window detached on 0.29.0, so upgraded to 0.29.2 and same behaviour. 

will uninstall and re-install to check

edit: checked and same result after reinstall of 0.29.2 via snap store or through sudo apt-get etc

**Comment by foldynl** [comment]
The good news is that I am able to reproduce it. You are probably using wayland. 

The bad news is that it looks like a bug in the external library (QT) that's used by QLog ([QT issue 87332](https://bugreports.qt.io/browse/QTBUG-87332) ). 

> checked and same result after reinstall of 0.29.2 via snap store or through sudo apt-get etc

Snap? Did you mean Flatpak. I don't see QLog  on Snapcraft. 

I'm not sure what you have and want to prefer installed (Flatpak or deb), but there are two ways to reattach the window. Both ways are workarounds that do not solve the issue in the QT library, but they just help you to reattach the window.

1) Flatpak

If you have installed QLog from Flathub then try to run it as below

`flatpak run --socket=x11 --nosocket=wayland --socket=fallback-x11 io.github.foldynl.QLog`

2) DEB package
There are two options:
a) try to run QLog via XOrg
choose XOrg session before login.
or b) you can remove QLog Window State. you will lose your window settings, but you will be able to redefined them again later.
edit a config file `~/.config/hamradio/QLog.conf` and remove a key `windowState` in section `General`. (make a backup of the config file before changing).


After this, you should be able to reattach the window as I showed in the Comment [2](https://github.com/foldynl/QLog/discussions/266#discussioncomment-7640973)

**Comment by foldynl** [comment]
Opened QLog issue #268 and Flatpak issue flathub/io.github.foldynl.QLog#3

**Reply by foldynl** [reply]
You should be able to attach the window if you are running Flatpak QLog. Just reinstall (uninstall and install) QLog from Flathub. Flatpak package is prepared without Wayland support.
